tests/auth/test_client.py

```diff
@@ -0,0 +1,267 @@
+    db, spawn_auth_client, delete_store, fake2, snapshot, mongo
+):
+    abs_client = await spawn_auth_client(permissions=[Permission.cancel_job])
+
+    if db == "mongo":
+
+        group = await fake2.groups.create()
+
+        await fake2.users.create()
+
+        user = await fake2.users.create()
+
+        await abs_client.add(GroupMembership(user.id, group.id, ["member"]))
+
+        assert await mongo.users.find({}, ["groups"]).to_list(None) == snapshot
+    else:
+        await abs_client.add(GroupMembership("ryanf", "sidney", ["member"]))
+
+        await abs_client.add(
+            GroupPermission(
+                "sidney", [Permission.cancel_job, Permission.modify_subtraction]
+            )
+        )
+
+        assert await abs_client.list_permissions("ryanf", "app", "virtool") == snapshot
+
+
+@pytest.mark.parametrize("db", ["mongo", "open_fga"])
+async def test_add_group_permissions(
+    delete_store, fake2, spawn_auth_client, mongo, db, snapshot
+):
+    abs_client = await spawn_auth_client()
+
+    group = await fake2.groups.create()
+    await fake2.groups.create()
+
+    await fake2.users.create()
+    await fake2.users.create(groups=[group])
+    await fake2.users.create(groups=[group])
+    await fake2.users.create(groups=[group])
+
+    if db == "mongo":
+
+        await abs_client.add(
+            GroupPermission(
+                group.id, [Permission.cancel_job, Permission.modify_subtraction]
+            )
+        )
+
+        assert (
+            await mongo.users.find({}, ["groups", "permissions"]).to_list(None)
+            == snapshot
+        )
+
+    else:
+
+        await write_tuple(
+            abs_client.open_fga, "user", "ryanf", "member", "group", "sidney"
+        )
+
+        await abs_client.add(
+            GroupPermission(
+                "sidney", [Permission.cancel_job, Permission.modify_subtraction]
+            )
+        )
+
+        assert await abs_client.list_permissions("ryanf", "app", "virtool") == snapshot
+
+
+async def test_add_user_permissions(delete_store, spawn_auth_client, snapshot):
+    abs_client = await spawn_auth_client(permissions=[Permission.cancel_job])
+
+    await abs_client.add(UserPermission("ryanf", [Permission.cancel_job]))
+
+    assert await abs_client.list_permissions("ryanf", "app", "virtool") == snapshot
+
+
+@pytest.mark.parametrize("db", ["mongo", "open_fga"])
+async def test_remove_group_permissions(
+    mongo, delete_store, db, spawn_auth_client, snapshot, fake2
+):
+    abs_client = await spawn_auth_client()
+
+    group = await fake2.groups.create()
+    await fake2.groups.create()
+
+    await fake2.users.create()
+    await fake2.users.create(groups=[group])
+    await fake2.users.create(groups=[group])
+    await fake2.users.create(groups=[group])
+
+    if db == "mongo":
+        await mongo.groups.find_one_and_update(
+            {"_id": group.id},
+            {"$set": {"permissions.cancel_job": True, "permissions.create_ref": True}},
+        )
```

Comment:
The document is returned by `find_one_and_update` so it is more expensive.
```suggestion
        await mongo.groups.update_one(
            {"_id": group.id},
            {"$set": {"permissions.cancel_job": True, "permissions.create_ref": True}},
        )
```